src/schema/v2/homeView/zones/legacy.ts

```diff
@@ -46,20 +47,41 @@ const LEGACY_ZONE_SECTIONS: HomeViewSection[] = [
   FeaturedFairs,
 ]
 
+/**
+ * Assemble the list of sections that can be displayed
+ */
 export async function getLegacyZoneSections(context: ResolverContext) {
-  const isAuthenticatedUser = !!context.accessToken
-
-  const displayableSections = LEGACY_ZONE_SECTIONS.reduce(
-    (sections, section) => {
-      const isDisplayable =
-        section.requiresAuthentication === false || // public content, or
-        (section.requiresAuthentication && isAuthenticatedUser) // user-specific content
+  const displayableSections: HomeViewSection[] = []
 
-      if (isDisplayable) sections.push(section)
-      return sections
-    },
-    [] as HomeViewSection[]
-  )
+  LEGACY_ZONE_SECTIONS.forEach((section) => {
+    if (isDisplayable(section, context)) {
+      displayableSections.push(section)
+    }
+  })
 
   return displayableSections
 }
+
+/**
+ * Determine if an individual section can displayed, consdering the current
+ * context, session, feature flags, etc.
+ */
+function isDisplayable(section: HomeViewSection, context: ResolverContext) {
+  // public content
+  const isPublicSection = section.requiresAuthentication === false
+
+  // personalized content
+  const isAuthenticatedUser = !!context.accessToken
+  const isValidPersonalizedSection =
+    section.requiresAuthentication && isAuthenticatedUser
+
+  // feature flagged sections
+  if (section.id === "home-view-section-featured-fairs") {
+    return isFeatureFlagEnabled(
```

Comment:
**musing / non-blocking**: Interesting that this check doesn't need to be async. We must be refreshing feature flags within Metaphysics at some regular interval, including any `context` configuration necessary, like a list of allowed user IDs.